### PR TITLE
Fix inaccurate validator balance metrics

### LINKED ISSUE #5305: Reported validator balances inconsistent
For my validator client (V0.3.9) the epoch summaries show a very inconsistent picture of the balance. The reported percent change is always negative (see #5247), but the absolute values are sometimes rising and sometimes falling. Also, the new balance of the previous epoch is not equal to the old balance of the current epoch.

Also the inclusion slot looks very strange to me, being larger than the current slot number.

```
Apr 04 09:16:48 ethnode-4ae769e8 validator[20051]: time="2020-04-04 09:16:48" level=info msg="Previous epoch voting summary" correctlyVotedHead=false correctlyVotedSource=false correctlyVotedTarget=false epoch=19211 inclusionDistance=18446744073709551615 inclusionSlot=18446744073709551615 newBalance=3.133435656 oldBalance=3.133479487 percentChange="-0.00140%" prefix=validator pubKey=0x8a8cd1076aaec9a4
...
Apr 04 09:23:12 ethnode-4ae769e8 validator[20051]: time="2020-04-04 09:23:12" level=info msg="Previous epoch voting summary" correctlyVotedHead=false correctlyVotedSource=false correctlyVotedTarget=false epoch=19212 inclusionDistance=18446744073709551615 inclusionSlot=18446744073709551615 newBalance=3.133510021 oldBalance=3.133550157 percentChange="-0.00128%" prefix=validator pubKey=0x8a8cd1076aaec9a4
...
Apr 04 09:29:36 ethnode-4ae769e8 validator[20051]: time="2020-04-04 09:29:36" level=info msg="Previous epoch voting summary" correctlyVotedHead=false correctlyVotedSource=false correctlyVotedTarget=false epoch=19213 inclusionDistance=18446744073709551615 inclusionSlot=18446744073709551615 newBalance=3.133345222 oldBalance=3.133390901 percentChange="-0.00146%" prefix=validator pubKey=0x8a8cd1076aaec9a4
```

### LINKED ISSUE #5600: Validator "Previous epoch voting summary" info message prints the wrong pubKey if two keys are in the same keymanager directory
# 🐞 Bug Report

### Description

 Validator "Previous epoch voting summary" info message prints the wrong pubKey if two keys are in the same directory. See the below logs for an example. 0xa7976999cafdfa24 is a key which is pending activation and is not valid yet. 0x949308ca28d2 is the key which is active and attesting, but it's not showing in the log message correctly. I discovered this while trying to make two pubkeys run on a single validator.

### Has this worked before in a previous version?

 Unknown

## 🔬 Minimal Reproduction

Create a validator keypair, fund it, wait for activation
Create a second validator keypair in another directory, then move the second keypair into the first keypair's directory.
Observe incorrect log messages

## 🔥 Error

<pre><code>

[2020-04-23 20:09:36]  INFO validator: Attestation schedule attesters=1 pubKeys=[0x949308ca28d2] slot=42048
[2020-04-23 20:16:00]  INFO validator: Previous epoch voting summary correctlyVotedHead=true correctlyVotedSource=true correctlyVotedTarget=true epoch=1314 inclusionDistance=1 inclusionSlot=41991 newBalance=64.020823481 oldBalance=64.02074911 percentChange=0.00012% pubKey=0xa7976999cafdfa24
</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
docker desktop for macOS
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
gcr.io/prysmaticlabs/prysm/beacon-chain version 7c986ca6f95d
gcr.io/prysmaticlabs/prysm/validator version 7194c01ac0e9

validator git commit version 49ca0751b6e37387b6df5a56d68980f93ccf377b
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Bug fix

**What does this PR do? Why is it needed?**
This PR solves the issue of validator metrics being inaccurate, previously the validator client was assuming the response from the beacon node was in the same order requested, but it is actually sorted by validator index. This changes the validator client to use the public key response to set the order received. 

Fixes #5305 and fixes #5600

### CODE REVIEW

File: validator/client/validator_metrics.go [hunk 1]
```diff
@@ -51,72 +51,64 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
...
-		pubKey := fmt.Sprintf("%#x", pkey[:8])
-		log := log.WithField("pubKey", pubKey)
-		fmtKey := fmt.Sprintf("%#x", pkey[:])
-		if missingValidators[bytesutil.ToBytes48(pkey)] {
-			if v.emitAccountMetrics {
-				validatorBalancesGaugeVec.WithLabelValues(fmtKey).Set(0)
-			}
-			continue
-		}
+	prevEpoch := (slot / params.BeaconConfig().SlotsPerEpoch) - 1
```
Comment (rauljordan) [inline validator/client/validator_metrics.go:65]: How about slot 0? I believe we have a helper func called helpers.PrevEpoch that checks for underflow

Comment (0xKiwi) [reply validator/client/validator_metrics.go:65]: This code doesn't run in epoch 0, but I'll use the helper func.

Comment (rauljordan) [reply validator/client/validator_metrics.go:66]: lets make a small helper then in this file that checks for underflow

Comment (0xKiwi) [reply validator/client/validator_metrics.go:66]: Added a comment to detail

**Comment by terencechain** [comment]
> A lot of changes in validator metrics. Note to self to cross check to ensure this doesn't break validator grafana dashboard

Testing now. Give me 30 mins

**Comment by rauljordan** [comment]
Just for the sake of defensive programming, always good to check for underflow @0xKiwi

**Pull request merged**